jgrapht-core/src/main/java/org/jgrapht/alg/scoring/AlphaCentrality.java

```diff
@@ -0,0 +1,280 @@
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor)
+    {
+        this(g, dampingFactor, EXOGENOUS_FACTOR_DEFAULT, MAX_ITERATIONS_DEFAULT, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactor the exogenous factor
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, double exogenousFactor)
+    {
+        this(g, dampingFactor, exogenousFactor, MAX_ITERATIONS_DEFAULT, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactormap the exogenous factor map
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, Map<V, Double> exogenousFactormap)
+    {
+        this(g, dampingFactor, exogenousFactormap, MAX_ITERATIONS_DEFAULT, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactor the exogenous factor
+     * @param maxIterations the maximum number of iterations to perform
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, double exogenousFactor, int maxIterations)
+    {
+        this(g, dampingFactor, exogenousFactor, maxIterations, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactormap the exogenous factor map
+     * @param maxIterations the maximum number of iterations to perform
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, Map<V, Double> exogenousFactormap, int maxIterations)
+    {
+        this(g, dampingFactor, exogenousFactormap, maxIterations, TOLERANCE_DEFAULT);
+    }
+
+    /**
+     * Create and execute an instance of AlphaCentrality.
+     * 
+     * @param g the input graph
+     * @param dampingFactor the damping factor
+     * @param exogenousFactor the exogenous factor
+     * @param maxIterations the maximum number of iterations to perform
+     * @param tolerance the calculation will stop if the difference of AlphaCentrality values between
+     *        iterations change less than this value
+     */
+    public AlphaCentrality(Graph<V, E> g, double dampingFactor, double exogenousFactor, int maxIterations, double tolerance)
+    {
+        this.g = g;
+        this.scores = new HashMap<>();
+
+        Map<V, Double> exofactor = new HashMap<>();
```

Comment:
I think the garbage man forgot to empty the trash bins today?

Reply:
Hehe.. Sorry!!